From 201ec8e75c7efc888441abf20efc2077f4fb3f7b Mon Sep 17 00:00:00 2001
From: AJMFactsheets <AJMFactsheets@gmail.com>
Date: Fri, 17 Jan 2020 17:17:54 -0600
Subject: [PATCH] Fix items not falling correctly

Since 1.14, Mojang has added an optimization which checks if items
should fall on every fourth tick.

However, Spigot's entity activation range class also has an
optimization which ticks active entities every fourth tick.
This can result in a state where an item will never properly fall
due to its move method never being called.

This patch resolves the conflict by ticking items every other tick,
since excluding item entities from Spigot's entity activation range
would offer worse performance overall.


diff --git a/src/main/java/net/minecraft/server/EntityItem.java b/src/main/java/net/minecraft/server/EntityItem.java
index e61af3f5..30a7843b 100644
--- a/src/main/java/net/minecraft/server/EntityItem.java
+++ b/src/main/java/net/minecraft/server/EntityItem.java
@@ -86,7 +86,7 @@ public class EntityItem extends Entity {
                 }
             }
 
-            if (!this.onGround || b(this.getMot()) > 9.999999747378752E-6D || (this.ticksLived + this.getId()) % 4 == 0) {
+            if (!this.onGround || b(this.getMot()) > 9.999999747378752E-6D || (this.ticksLived + this.getId()) % 2 == 0) { // Paper - Check for item movement twice as often to not conflict with Spigot's Entity Activation Range
                 this.move(EnumMoveType.SELF, this.getMot());
                 float f = 0.98F;
 
-- 
2.17.1

